code/modules/events/disease_outbreak.dm

```diff
@@ -9,13 +9,28 @@
 		D = new virus_type()
 	else
 		var/datum/disease/advance/A = new /datum/disease/advance
+		// Try generating a random spreading virus. If that fails just use the last generated one
+		for(var/i in 1 to 100)
+			var/symptom_amount = rand(2, 6)
+			QDEL_LIST(A.symptoms)
+			A.symptoms = A.GenerateSymptoms(1, 6, symptom_amount)
+			A.Refresh(FALSE, FALSE) // Don't actually archive the miscreants
+			if(VirusCanSpread(A))
+				break
 		A.name = capitalize(pick(GLOB.adjectives)) + " " + capitalize(pick(GLOB.nouns + GLOB.verbs)) // random silly name
-		A.GenerateSymptoms(1,9,6)
-		A.AssignProperties(list("resistance" = rand(0,11), "stealth" = rand(0,2), "stage_rate" = rand(0,5), "transmittable" = rand(0,5), "severity" = rand(0,10)))
+		A.Refresh() // Actually archive it with it's name
 		D = A
 
+	if(istype(D, /datum/disease/advance))
+		var/datum/disease/advance/A = D
+		A.event_spawned = TRUE // Will make random advanced viruses and default ones like the flu or such also able to spread
 	D.carrier = TRUE
 
+/datum/event/disease_outbreak/proc/VirusCanSpread(datum/disease/advance/A)
```

Comment:
First, I'm not sure all viruses are spreadable by default. What about viruses with no spreading flags? You don't handle that case.
Second, why are blood-only virusses spread by sneezing?
Third, why not just add an extra symptom rather than re-genning until you get the one you want?

Reply:
All viruses are atleast spread by blood. That is how advanced viruses work.

Smeezing wont force the airborne trait on the virus. Instead it makes the virus airborne due to the activation of the symptom. Sneezing when activated will spread the virus to surrounding mobs. Toxic filterings activation will heal toxins.

Adding one random symptom can make the virus get into a minimum it cant get out of. The more symptoms a virus has the lower its transmition stat is. Rerolling the entire virus will thus result in better results